_posts/2021-05-14-bring-gitops-to-your-functions-with-argocd-part-2-the-missing-piece-tekton.md

```diff
@@ -0,0 +1,612 @@
+task.tekton.dev/git-clone created
+task.tekton.dev/kaniko created
+task.tekton.dev/deploy-using-kubectl created
+
+$ tkn tasks list
+NAME              DESCRIPTION              AGE
+faas-cli-build    This task create a ...   53 seconds ago
+git-clone         These Tasks are Git...   53 seconds ago
+kaniko            This Task builds so...   53 seconds ago
+update-manifest                            53 seconds ago
+
+$ kubectl apply -f pipeline/build-and-deploy-pipeline.yaml
+pipeline.tekton.dev/build-and-deploy created
+
+$ tkn pipeline list
+NAME               AGE              LAST RUN   STARTED   DURATION   STATUS
+build-and-deploy   17 seconds ago   ---        ---       ---        ---
+
+$ kubectl apply -f triggers/
+eventlistener.triggers.tekton.dev/github-listener-interceptor created
+triggerbinding.triggers.tekton.dev/github-pr-binding created
+triggertemplate.triggers.tekton.dev/github-template created
+serviceaccount/tekton-triggers-example-sa created
+role.rbac.authorization.k8s.io/tekton-triggers-example-minimal created
+rolebinding.rbac.authorization.k8s.io/tekton-triggers-example-binding created
+clusterrole.rbac.authorization.k8s.io/tekton-triggers-example-clusterrole created
+clusterrolebinding.rbac.authorization.k8s.io/tekton-triggers-example-clusterbinding created
+
+$ tkn triggertemplate list
+NAME              AGE
+github-template   31 seconds ago
+
+$ tkn triggerbinding list
+NAME                AGE
+github-pr-binding   34 seconds ago
+```
+
+Verify if everything is working before moving onto the next step.
+
+```bash
+$ kubectl get pods
+NAME                                              READY   STATUS             RESTARTS   AGE
+el-github-listener-interceptor-7bc945b898-jlst8   1/1     Running            0          68s
+hellofunction-7849c4cf8d-zdgm4                    0/1     InvalidImageName   0          52m
+```
+
+_el-github-listener-interceptor-7bc945b898-jlst8_ this Pod is our _EventListener_, we should open it to the internet to be able to get events from the Github, now we'll do port-forwarding to make it reachable from localhost, then we'll run _inlets-pro_ to make it reachable from the internet.
+
+But before doing that we need create some sort of secret for Github Webhook. Save this secret because we'll use that when we set up Webhook in Github for our repository.
+
+```bash
+$ export TEKTON_TUTORIAL_SECRET_TOKEN=${TEKTON_TUTORIAL_SECRET_TOKEN-$(head -c 24 /dev/random | base64)}
+$ kubectl create secret generic github-secret --from-literal=secretToken=$TEKTON_TUTORIAL_SECRET_TOKEN
+$ echo "TEKTON_TUTORIAL_SECRET_TOKEN: $TEKTON_TUTORIAL_SECRET_TOKEN"
+xxxxx
+```
+
+Let's make it reachable our EventListener.
+
+```bash
+$ kubectl port-forward svc/el-github-listener-interceptor 8080
+Forwarding from 127.0.0.1:8080 -> 8000
+Forwarding from [::1]:8080 -> 8000
+
+$ inlets-pro http client --token=$TOKEN --url=$WSS --upstream http://127.0.0.1:8080 --license-file $HOME/.inlets/LICENSE --auto-tls=false
```

Comment:
I don't understand how you created the inlets server? Have you skipped over all that part?